### ISSUE TITLE
[ML Classifier] Implement ML Classifier (XGBoost)

### ISSUE THREAD

**Issue by zerith-jag** [JustAGameZA/coding-agent#101]
## Overview
Implement machine learning classifier using XGBoost for improved accuracy (95% accuracy, 50ms latency).

## Tasks
- [ ] Create `MLClassifier` class in `domain/classifiers/`
- [ ] Implement feature extraction:
  - TF-IDF vectorization (max 1000 features)
  - Task description length
  - Keyword presence (binary features)
  - Code snippet detection (regex patterns)
- [ ] Load pre-trained XGBoost model from disk
- [ ] Implement prediction with probability scores
- [ ] Map predictions to task types and complexity
- [ ] Add model version tracking
- [ ] Implement model warm-up on startup
- [ ] Write unit tests with sample data
- [ ] Add performance benchmarks

## Model Pipeline
```python
class MLClassifier:
    def __init__(self):
        self.vectorizer = load_vectorizer()
        self.model = load_xgboost_model()
    
    async def classify(self, request: ClassificationRequest):
        features = self.extract_features(request.task_description)
        prediction = self.model.predict_proba(features)
        return ClassificationResult(...)
```

## Confidence Thresholds
- High confidence: ≥ 0.70 (use ML result)
- Low confidence: < 0.70 (fallback to LLM)

## Acceptance Criteria
- 95% accuracy on validation set
- Average latency < 50ms
- Handles 14% of traffic (medium confidence cases)
- Unit test coverage ≥ 85%

## Priority
**High** - Critical accuracy improvement

## Phase
Phase 2 - Week 11-12: ML Classifier Service

## Dependencies
- Pre-trained model (can use dummy model initially)
